File: capstone/src/main/java/impact/capstone/service/UserService.java
package impact.capstone.service;

@Service
public class UserService {

    @Autowired
    private UserRepository userRepository;

    public void signUp(UserDto userDto) {
        User user = new User();
        user.setUsername(userDto.getUsername());
        user.setPassword(userDto.getPassword());
        userRepository.save(user);
    }
}

File: server/src/main/kotlin/deu/capstone/impact/server/service/UserService.kt
package deu.capstone.impact.server.service

import deu.capstone.impact.server.dto.user.LoginRequestDto
import deu.capstone.impact.server.entity.user.User
import deu.capstone.impact.server.repository.UserRepository
import org.springframework.stereotype.Service

@Service
class UserService(
        val userRepository: UserRepository,
) {
    // 로그인을 위한 메서드
    fun login(data: LoginRequestDto): User {
        val user = userRepository.findByEmail(data.email) ?: error("")
        check(user.password == data.password)
        return user
    }
}